sql/src/main/java/io/crate/operation/projectors/ShardRequestAccumulator.java

```diff
@@ -95,8 +108,13 @@ public int batchSize() {
         if (currentRequest.itemIndices().isEmpty() && isLastBatch) {
             return CompletableFuture.completedFuture(getSingleRowWithRowCount());
         }
+
+        final String nodeId = getLocalNodeId();
```

Comment:
would it make sense to require the localNodeId in the constructor and do this outside?